### DISCUSSION TITLE
How to create routes with path parameters to custom pages?

### DISCUSSION THREAD

**Discussion by mxc** [filamentphp/filament#9012] [Help] [answered]
Hi there,

I have created a custom page with 

`php artisan make:filament-page Finance`

I want to pass an id in the url e.g /admin/finance/123. I have overridden the getSlug method:

```
    public static function getSlug(): string
    {
        return 'finance/{id}';
    }
```

When accessing the page with /admin/finance/123 I get:

`Missing required parameter for [Route: filament.admin.pages.finance.{id}] [URI: admin/finance/{id}] [Missing parameter: id].`

I have tried adding a route in web.php. I have tried overriding "    public static function getRouteName(?string $panel = null): string". No matter what I try it seems impossible to get the page generated without it complaining about the missing parameter id. Please advise. thanks

**Comment by elbokh** [comment, marked as answer]
Hi there!

To avoid getting the error make sure you add: 

`protected static bool $shouldRegisterNavigation = false;`

Because it will try to load the page when it's registered in the navigation with **null** as the ID.


Your code should look like this:

```
use App\Models\Finance;


class FinancePage extends Page
{
	protected static string $view = 'filament.pages.finance';

	protected static ?string $slug = 'finance/{id}';
	protected static bool $shouldRegisterNavigation = false;
	public Finance $record; # Assuming Finance is a model.

	public function mount($id): void
	{

		$this->record = (new Finance)->FindOrFail($id);

	}
}
```

Hope this helps ! 😊

**Reply by mxc** [reply]
Thanks so much. That did the trick.

**Reply by vaishalir04** [reply]
Got two pages i passed a parameter in ViewAction  of page SaleByCustomer Tables\Actions\ViewAction::make()
                            ->url(fn (Voucher $record) => route('filament.admin.pages.sale-by-customer-view', $record->ledger_id)),                 
 i am passing legder id as parameter how to access that in SaleByCustomerView page

**Reply by denizaygundev** [reply]
@elbokh this worked for me too, thank you! Wish it was in the filament docs.

**Comment by MusheAbdulHakim** [comment]
In **V3**, you can do this using:

```php
    protected static ?string $slug = 'finance/{id?}';
```

Or in your case, make the parameter optional like:

```php
    public static function getSlug(): string
    {
        return 'finance/{id?}';
    }
```
